### PR TITLE
bundle validate: validate a bundle directory

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:** validate a directory of manifests by passing a directory as an arg nstead of an image tag.

**Motivation for the change:** So you don't have to build and push an image before validating its eventual contents.

TODO:
- [x] ~discuss CLI flags to make sure we want to make this breaking change~
- [x] update CLI and usage docs
- [x] add command test case ~will do in #2715 since this fixes the bundle image builder~

/kind feature

### CODE REVIEW

File: cmd/operator-sdk/bundle/validate.go [hunk 1]
```diff
@@ -15,89 +15,156 @@
...
+	return nil
+}
+
+func (c *bundleValidateCmd) addToFlagSet(fs *pflag.FlagSet) {
+	fs.StringVarP(&c.directory, "directory", "d", "",
+		"The directory where bundle manifests and metadata are located or, "+
+			"if --content-only is set, the path to a bundle. Either --directory or --tag must be set but not both")
+	fs.StringVarP(&c.imageTag, "tag", "t", "",
+		"The path of a registry to pull from, image name and its tag that present the bundle image "+
+			"(e.g. quay.io/test/test-operator:latest). Either --directory or --tag must be set but not both")
```
Comment (joelanford) [inline cmd/operator-sdk/bundle/validate.go]: WDYT about just keeping the CLI the same and doing the following psuedocode?
```
if argv[0] is a path to a local directory {
    validate argv[0] as bundle directory
} else {
    validate argv[0] as image
}
```

File: cmd/operator-sdk/bundle/validate.go [hunk 2]
```diff
@@ -15,89 +15,156 @@
...
-		"Tool to extract container images. One of: [docker, podman]")
+	// Validate bundle format if --content-only=false.
+	if c.contentOnly {
+		logger.Info("Validating bundle contents")
+	} else {
+		logger.Info("Validating bundle format and contents")
+		if err = val.ValidateBundleFormat(dir); err != nil {
+			logger.Fatalf("Bundle format validation failed: %v", err)
+		}
+	}
```
Comment (joelanford) [inline cmd/operator-sdk/bundle/validate.go]: Any reason we don't always want to validate the bundle format as well?

File: cmd/operator-sdk/bundle/validate.go (hunk 1 above)
Comment (estroz) [reply cmd/operator-sdk/bundle/validate.go]: I was thinking we could do this with `[docker|podman] image -q <image-tag>`. I'll code it out.

Comment (joelanford) [reply cmd/operator-sdk/bundle/validate.go]: It’s probably important that the directory path use case does not require docker or podman or any other image builder to be present. I’m thinking of using this in a CI or pipeline scenario where operator-sdk is running in an unprivileged container.

File: cmd/operator-sdk/bundle/validate.go (hunk 2 above)
Comment (estroz) [reply cmd/operator-sdk/bundle/validate.go]: With the changes in in #2715 this isn't necessary anymore. The original idea was to at least allow validation of a bundle directory even though `bundle create` was broken (see #2662).

**Comment by estroz** [comment]
Will rebase on top of #2746 and add test cases.

/hold

File: doc/user/olm-catalog/bundle-cli.md [hunk 3]
```diff
@@ -17,6 +17,7 @@ The following `operator-sdk` subcommands create or interact with Operator on-dis
   | **operator-sdk** | **opm** |
   |--- |--- |
   | `operator-sdk bundle validate <image-tag>` | `opm alpha bundle validate --tag <image-tag>` |
+  | `operator-sdk bundle validate <directory>` | no equivalent |
```
Comment (camilamacedo86) [inline doc/user/olm-catalog/bundle-cli.md:21]: This update is missing in the website

**Pull request merged**